nautobot_ssot_infoblox/diffsync/adapters/infoblox.py

```diff
@@ -24,21 +25,28 @@ class InfobloxAdapter(DiffSync):
 
     top_level = ["prefix", "ipaddress", "vlangroup", "vlan"]
 
-    def __init__(self, *args, job=None, sync=None, **kwargs):
+    def __init__(self, *args, job=None, sync=None, conn=None, **kwargs):
         """Initialize Infoblox.
 
         Args:
             job (object, optional): Infoblox job. Defaults to None.
             sync (object, optional): Infoblox DiffSync. Defaults to None.
+            conn (object, optional): InfobloxAPI connection. Defaults to None.
         """
         super().__init__(*args, **kwargs)
         self.job = job
         self.sync = sync
-        self.conn = InfobloxApi()
+        self.conn = conn
         self.subnets = []
 
+        if self.conn in [None, False]:
+            self.job.log_failure(
+                message="Improperly configured settings for communicating to Infoblox. Please validate accuracy."
+            )
+            raise PluginImproperlyConfigured
```

Comment:
This will only work as long as we keep the plugin settings in `nautobot_config.py`. I'm thinking we should move the instance definitions into database objects to allow for multiple end-points to be used with.

Reply:
Correct. It is pre 1.0, but do we have that ability today and want to migrate to that? We can do in a diff PR, but not sure if it's possible to store that info in the database right now?

Reply:
That was my intention, to make that change in a future PR once I address #43.